packages/jsdoc-plugins/lib/relation-fixer/getmissingdocletsdata.js

```diff
@@ -62,57 +72,62 @@ function getMissingDocletsData( docletCollection, childDoclet, options ) {
 	};
 }
 
-// Creates a <longname, doclet> map.
-//
-// @param {DocletCollection} doclets
-// @returns {Object}
-function createDocletMap( doclets ) {
-	const map = {};
-
-	for ( const doclet of doclets.getAll() ) {
-		map[ doclet.longname ] = doclet;
-	}
-
-	return map;
-}
-
-// Gets doclets from entities related to current doclet (e.g. implemented by it)
-// and matching criteria given in options.filter.
-function getDocletsToAdd( docletCollection, childDoclet, options = {} ) {
+/**
+ * Gets doclets from entities related to current doclet (e.g. implemented by it)
+ * and matching criteria given in options.filter.
+ *
+ * @param {DocletCollection} docletCollection
+ * @param {Readonly<Doclet>} childDoclet
+ * @param {Object} options
+ * @param {'augmentsNested'|'mixesNested'|'implementsNested'} options.relation
+ * @param {Partial<Doclet>} [options.filter] An object used to filter missing doclets (e.g. { scope: 'static' }).
+ */
+function getDocletsToAdd( docletCollection, childDoclet, options ) {
 	if ( !isNonEmptyArray( childDoclet[ options.relation ] ) ) {
 		return [];
 	}
 
 	// Longnames of doclets which are related ( extended, mixed, implemented ) to childDoclet.
 	const ancestors = childDoclet[ options.relation ];
 
-	return ancestors.reduce( ( docletsToAdd, longname ) => {
-		const toAdd = docletCollection.get( `memberof:${ longname }` ).filter( d => {
-			let isMatchingFilterOptions = true;
+	const docletToAdd = ancestors.reduce( ( /** @type {Array.<Doclet>} */ docletsToAdd, longname ) => {
+		const toAdd = docletCollection.get( `memberof:${ longname }` ).filter( doclet => {
+			let matchingFilterOptions = true;
+
 			// Filter out ignored, inherited, undocumented.
-			const isUnwanted = d.ignore === true ||
-				d.undocumented === true ||
-				d.inheritdoc !== undefined;
+			if ( doclet.ignore ||
+				doclet.undocumented ||
+				typeof doclet.inheritdoc == 'string'
+			) {
+				return false;
+			}
 
 			for ( const key of Object.keys( options.filter || {} ) ) {
-				if ( d[ key ] !== options.filter[ key ] ) {
-					isMatchingFilterOptions = false;
+				if ( doclet[ key ] !== options.filter[ key ] ) {
+					matchingFilterOptions = false;
 				}
 			}
 
-			return isMatchingFilterOptions && !isUnwanted;
+			return matchingFilterOptions;
 		} );
 
-		docletsToAdd.push( ...toAdd );
-
-		return docletsToAdd;
+		return [
+			...docletsToAdd,
+			...toAdd
+		];
 	}, [] );
+
+	return docletToAdd;
 }
 
 function isNonEmptyArray( obj ) {
 	return Array.isArray( obj ) && obj.length > 0;
 }
 
+/**
+ * @param {Readonly<Doclet>} parentDoclet
```

Comment:
What is the `Readonly<Doclet>`?

Reply:
Readonly<T> is an object of type T that can't be modified. But if you wish I can remove that.